### PR TITLE
Analyser: SAST Fixes

### PR DESCRIPTION
**Pull request by kingthorin**
- Remove unnecessary comments, and address various SAST (SonarLint) issues.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: zap/src/main/java/org/parosproxy/paros/core/scanner/Analyser.java [hunk 1]
```diff
@@ -508,12 +499,8 @@ public boolean isFileExist(HttpMessage msg) {
...
                                 .toString()
                                 .replaceAll(getPathRegex(uri), "")
-                                .replaceAll("\\s[012]\\d:[0-5]\\d:[0-5]\\d\\s", "");
-                // ZAP: FindBugs fix - added call to HttpBody.toString()
-                if (sample.getMessage().getResponseBody().toString().equals(body)) {
-                    return false;
-                }
-                return true;
+                                .replaceAll(PATTERN_THREE_SEGMENT_TIME, "");
+                return sample.getMessage().getResponseBody().toString().equals(body);
```
Comment (thc202) [inline zap/src/main/java/org/parosproxy/paros/core/scanner/Analyser.java:503]: Needs to be negated.

File: zap/src/main/java/org/parosproxy/paros/core/scanner/Analyser.java [hunk 2]
```diff
@@ -141,7 +141,7 @@ private void addAnalysedHost(URI uri, HttpMessage msg, int errorIndicator) {
             mapVisited.put(uri.toString(), new SampleResponse(msg, errorIndicator));
 
         } catch (HttpMalformedHeaderException | DatabaseException e) {
-            logger.error("Failed to persist the message: " + e.getMessage(), e);
+            logger.error("Failed to persist the message: {}", e.getMessage(), e);
```
Comment (thc202) [inline zap/src/main/java/org/parosproxy/paros/core/scanner/Analyser.java:144]: These are being done in the other PR, should not conflict though.

Comment (kingthorin) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/Analyser.java:144]: Ya, I’ll sort it out whichever lands first.

**Comment by kingthorin** [comment]
Fixed

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**